Lib/test/test_dataclasses/__init__.py

```diff
@@ -1524,6 +1524,21 @@ class A(types.GenericAlias):
         self.assertTrue(is_dataclass(type(a)))
         self.assertTrue(is_dataclass(a))
 
+    def test_dataclass_inheritance(self):
+        @dataclass
+        class X:
+            y: int
+
+        class Z(X):
+            pass
+
+        # Check if X is a dataclass
+        self.assertTrue(is_dataclass(X), "X should be a dataclass")
+        # Check if Z is a dataclass, it should inherit the dataclass behavior from X
+        self.assertTrue(is_dataclass(Z), "Z should be a dataclass because it inherits from X")
+        # Create an instance of Z and check if the default values are set correctly
+        z_instance = Z(y=5)
+        self.assertTrue(is_dataclass(z_instance), "z_instance should be a dataclass because it is an instance of Z")
```

Comment:
```suggestion
        self.assertTrue(is_dataclass(X), "X should be a dataclass")
        self.assertTrue(
            is_dataclass(Z),
            "Z should be a dataclass because it inherits from X",
        )
        z_instance = Z(y=5)
        self.assertTrue(
            is_dataclass(z_instance),
            "z_instance should be a dataclass because it is an instance of Z",
        )
```